### ISSUE TITLE
loss_D about wgan

### ISSUE THREAD

**Issue by jeejeeli** [eriklindernoren/PyTorch-GAN#17]
Hi:
    Thanks for sharing your impressive repo, I found the loss_D in wan.py is :
`loss_D = -torch.mean(discriminator(real_imgs)) + torch.mean(discriminator(fake_imgs))` ,but the wgan's paper described the loss_D as follow:
![image](https://user-images.githubusercontent.com/19733142/41358164-65a6b210-6f5a-11e8-8418-b45158c53389.png)

So is it a bug or other reasons？

**Comment by jeejeeli** [comment]
I fix  loss_D = torch.mean(discriminator(real_imgs)) - torch.mean(discriminator(fake_imgs)) ,and then doing training, I found the generated image is black number and white background, but when I fix  sign of loss_D and loss_G from your original code,The generated image looks like is normal（white number and black background），becauese I am rookie about GAN,It make me very confused,I hope you can help me,thanks

**Comment by jeejeeli** [comment]
I have understood the loss_D,but in my personal opinion,the  loss_D should be wrote as follow:
loss_D  = -torch.mean(discriminator(real_imgs)) + torch.mean(discriminator(fake_imgs)) -1
is it right ?

**Comment by eriklindernoren** [comment]
I believe that since they add the gradient to the weights during update they perform gradient ascent instead of descent, and because of this in order to do gradient descent you will have to multiply the gradient with minus one. Because of this the loss will be:

```python
-1 * (torch.mean(discriminator(real_imgs)) - torch.mean(discriminator(fake_imgs)) = 
-torch.mean(discriminator(real_imgs)) + torch.mean(discriminator(fake_imgs))
```

**Comment by jeejeeli** [comment]
@eriklindernoren  thank you
